zrml/orderbook-v1/src/lib.rs

```diff
@@ -334,7 +335,13 @@ pub struct Order<AccountId, Balance, MarketId> {
 }
 
 impl<AccountId, Balance: CheckedSub + CheckedMul, MarketId> Order<AccountId, Balance, MarketId> {
-    pub fn cost(&self) -> Balance {
-        self.total.checked_sub(&self.filled).unwrap().checked_mul(&self.price).unwrap()
+    pub fn cost(&self) -> Result<Balance, DispatchError> {
+        match self.total.checked_sub(&self.filled) {
+            Some(subtotal) => match subtotal.checked_mul(&self.price) {
+                Some(cost) => Ok(cost),
+                _ => Err(DispatchError::Arithmetic(ArithmeticError::Overflow)),
+            },
+            _ => Err(DispatchError::Arithmetic(ArithmeticError::Overflow)),
```

Comment:
(pedantic) Would be nice to use the correct `ArithmeticError::Underflow` here
```suggestion
            _ => Err(DispatchError::Arithmetic(ArithmeticError::Underflow)),
```